File: 1441 Build an Array With Stack Operations/1441build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<int> a;
        vector<string> s;
        int c1 = 0; int c2=0;
        int i=1;
        while(target!=a){
            a.push_back(i);
            s.push_back("Push");
            if(a[c1] == target[c2]){
                c1++;c2++;
            }
            else{
                s.push_back("Pop");
                a.pop_back();
            }
            i++;
        }
        return s;
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        cur_ele = 1
        i = 0
        ans = []

        while cur_ele <= n and i<len(target):
            if target[i] == cur_ele:
                ans.append("Push")
                i += 1
            else:
                ans.append("Push")
                ans.append("Pop")

            cur_ele += 1
        
        return ans